src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/SensorsVisualizer.java

```diff
@@ -0,0 +1,56 @@
+package hu.oe.nik.szfmv.automatedcar.systemcomponents;
+
+import hu.oe.nik.szfmv.automatedcar.bus.VirtualFunctionBus;
+import hu.oe.nik.szfmv.automatedcar.bus.packets.input.InputPacket;
+import hu.oe.nik.szfmv.automatedcar.input.InputHandler;
+
+
+
+public class SensorsVisualizer extends SystemComponent {
+
+    private InputHandler inputHandler;
+    private InputPacket inputPacket;
+    private boolean radarVizualizerPressed;
+    private boolean cameraVizualizerPressed;
+    private boolean ultrasonicVizualizerPressed;
+
+    /**
+     * @param virtualFunctionBus VirtualFunctuonBus parameter
+     */
+    public SensorsVisualizer(VirtualFunctionBus virtualFunctionBus) {
+        super(virtualFunctionBus);
+
+        inputPacket = InputPacket.getInstance();
+        virtualFunctionBus.inputPacket = inputPacket;
+        inputHandler = InputHandler.getInstance();
+        radarVizualizerPressed = false;
+        cameraVizualizerPressed = false;
+        ultrasonicVizualizerPressed = false;
+    }
+
+    @Override
+    public void loop() {
+        if (inputHandler.isRadarTestPressed()) {
+            inputPacket.setRadarVizualizerState(state((radarVizualizerPressed)));
+        }
+        if (inputHandler.isCameraTestPressed()) {
+            inputPacket.setCameraVizualizerState(state(cameraVizualizerPressed));
+        }
+        if (inputHandler.isUltrasonicTestPressed()) {
+            inputPacket.setUltrasonicVizualizerState(state(ultrasonicVizualizerPressed));
+        }
+    }
+
+    private boolean state(boolean pressed) {
+        boolean state = false;
+        if (!pressed) {
+            state = true;
+        } else {
+            state = false;
```

Comment:
Ez az ág minimum redundáns. Közvetve az IDEA is figyelmeztet erre (az if-re teszi a sárgított figyelmeztetést)! Ha elfogadod az egyszerűsítési javaslatait, akkor így fog kinézni:
```java
    private boolean state(boolean pressed) {
        boolean state;
        state = !pressed;
        return state;
    }
```
Amely igazából tovább egyszerűsíthető:
```java
    private boolean state(boolean pressed) {
        return !pressed;
    }
```
Amely felveti a kérdést, hogy minek is van szükség erre a metódusra, hiszen följebb lehetne írni azt, hogy `!radarVizualizerPressed`, `!cameraVizualizerPressed` és `!ultrasonicVizualizerPressed`.

Reply:
Javítottam, kivettem az elágazást.